Embrace pystream

RealOrangeOne opened the pull request:
This PR replaces `srcomp-stream` with `srcomp-pystream` completely. 

The canonical spelling for the URL remains `/stream`, however `/pystream` (with or without trailing `/`) still exists and will redirect. At some point in the future, `/pystream` can be removed, but for now it's no additional maintenance to keep this around.

I've tested locally that the configuration applied properly, that `pystream` starts correctly, that it's reachable at `/stream` and that `/pystream/` redirects as expected.

RealOrangeOne committed:
Replace the legacy stream with pystream

Both sets of URLs remain valid, in case other clients haven't migrated

WillB97 commented on modules/compbox/templates/compbox-index.html.erb:
```diff
@@ -28,8 +28,7 @@
             <li><a href="/comp-api/">Comp API</a></li>
-            <li><a href="/stream">Comp API Stream (legacy)</a></li>
-            <li><a href="/pystream/">Comp API Stream (experimental)</a></li>
+            <li><a href="/stream/">Comp API Stream</a></li>
```
Should this be `/stream`?
```suggestion
            <li><a href="/stream">Comp API Stream</a></li>
```

RealOrangeOne replied:
We should probably decide on what the canonical URL should be. Personally I prefer URLs with trailing slashes. Whatever we choose, we should be consistent.

WillB97 commented on modules/compbox/templates/nginx.conf.erb:
```diff
@@ -32,25 +32,8 @@ server {
-    }
-
-    location = /pystream {
-        add_header Access-Control-Allow-Origin *;
```
You may still need this header to avoid CORS issues in a few places

RealOrangeOne committed:
Add CORS headers for stream

RealOrangeOne committed:
Remove trailing slash from stream URL on compbox homepage

RealOrangeOne committed:
Simplify location blocks for stream

Have a dedicated block for the fallback and redirect, rather than merging.

nginx also complains about using a regex with `proxy_pass`
